halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/service/distributed/google/GoogleDistributedService.java

```diff
@@ -134,9 +134,13 @@ default String getStartupScript() {
   default String getArtifactId(String deploymentName) {
     String artifactName = getArtifact().getName();
     String version = getArtifactService().getArtifactVersion(deploymentName, getArtifact());
-    return String.format("projects/%s/global/images/%s",
-        getGoogleImageProject(deploymentName),
-        String.join("-", "spinnaker", artifactName, version.replace(".", "-").replace(":", "-")));
+    if (version == null) {
```

Comment:
I think this should be handled upstream from here. I realize this is where the exception will be thrown,  but I think the deployer in charge of picking which services to create should be smarter about asking for "artifact ids" of services that can't be created.

Reply:
I'm sort of torn on whether this should be handled here or upstream.  I actually think that it's this function (or perhaps the downstream one `getArtifactVersion`) that's in the best position to know whether or not a given service should be built.  Effectively, the question of whether a service should be built is answered by looking at whether it's in the BOM---which is what this function is doing to get the version to deploy.  So I think it make sense that asking "what version of X should I deploy?" would return `null` if the answer is not to deploy X.

That being said, I agree that it's a bit silly that we still end up with an entry in `endpoints.services` for a service that's not being deployed...so maybe we should write a `boolean serviceExists()` that just checks if there's a version at all and short-circuit all the settings-building if it's false.  Let me see what this last idea would look like.